src/Sprites/spriteMap.ts

```diff
@@ -0,0 +1,627 @@
+                this.changeTiles(this.editorLayer, pos, this.editorTile);
+            };
+
+            let _drawObs: Nullable<Observer<Scene>> = null;
+
+            this._output.actionManager.registerAction(new ExecuteCodeAction(
+                ActionManager.OnLeftPickTrigger,
+                (evt: any) => {
+                        if (((this._scene) && this._scene.activeCamera)) {
+                        if (this._scene.getEngine && this._scene.getEngine().getInputElement) {
+                            let canvas: Nullable<HTMLElement> = this._scene.getEngine().getInputElement();
+                            if (!canvas) {
+                                return;
+                            }
+                            this._scene.activeCamera.detachControl(canvas);
+                            _drawObs = this._scene.onAfterRenderObservable.add(drawing);
+                        }
+                    }
+                })
+            );
+
+            this._output.actionManager.registerAction(new ExecuteCodeAction(
+                ActionManager.OnPickUpTrigger,
+                (evt: any) => {
+                    if (((this._scene) && this._scene.activeCamera)) {
+                        if (this._scene.getEngine && this._scene.getEngine().getInputElement) {
+                            let canvas: Nullable<HTMLElement> = this._scene.getEngine().getInputElement();
+                            if (!canvas) {
+                                return;
+                            }
+                            this._scene.activeCamera.attachControl(canvas);
+                            this._scene.onAfterRenderObservable.remove(_drawObs);
+                            this.editorPos = new Vector2(-1, -1);
+                        }
+                    }
+                })
+            );
+
+            if (!this._scene.actionManager) {
+                 this._scene.actionManager = new ActionManager(this._scene);
+            }
+
+            this._scene.actionManager.registerAction(
+                new ExecuteCodeAction(ActionManager.OnKeyDownTrigger, (evt: any) => {
+                    switch (evt.sourceEvent.key){
+                        case '-':
+                            this.editorTile = Math.max(0.0, this.editorTile - 1) ;
+                        break;
+                        case '=':
+                            this.editorTile = Math.min(this.editorTile + 1, this.spriteCount);
+                        break;
+                        case '[':
+                            this.editorLayer = Math.max(0.0, this.editorLayer - 1) ;
+                        break;
+                        case ']':
+                            this.editorLayer = Math.min(this.editorLayer + 1, this.options.layerCount);
+                        break;
+                    }
+                })
+            );
+
+        }else {
+            obfunction = () => {
+                this._time += this._scene.getEngine().getDeltaTime() * 0.01;

```

Comment:
why 0.01?

Reply:
It just made it smooth for the animations it seemed like.

 I just deleted it they have access to a speed variable when the animations are created.